fashion-messaging-app-ui/src/RequireLogin.jsx

```diff
@@ -0,0 +1,14 @@
+import { Navigate, useLocation } from 'react-router-dom';
+import { useContext } from 'react';
+import { UserContext } from './UserContext';
+
+export default function RequireLogin({ children }) {
+  const { user } = useContext(UserContext);
+  const location = useLocation();
+ 
+  if (!user) {
+    return <Navigate to="/login" state={{ from: location }} replace />
+}
```

Comment:
This means if the user is not logged in, it should route to the login route and the user cannot be redirected to anywhere in the website until they are authenticated